### PR TITLE
feat: support openssl3

### PR DESCRIPTION
**Pull request by AlinsRan**
### Description

It is understood that OPENSSL 1.1.1 has already been [EOL](https://www.openssl.org/blog/blog/2023/03/28/1.1.1-EOL/), so we will upgrade to openssl 3.

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: .github/workflows/cli-master.yml [hunk 1]
```diff
@@ -6,11 +6,6 @@ on:
     paths-ignore:
       - 'docs/**'
       - '**/*.md'
-  pull_request:
-    branches: [master]
-    paths-ignore:
-      - 'docs/**'
-      - '**/*.md'
```
Comment (monkeyDluffy6017) [inline .github/workflows/cli-master.yml:13]: Why delete this?

File: ci/common.sh [hunk 2]
```diff
@@ -25,7 +25,8 @@ export_or_prefix() {
     export OPENRESTY_PREFIX="/usr/local/openresty"
     export APISIX_MAIN="https://raw.githubusercontent.com/apache/apisix/master/apisix-master-0.rockspec"
     export PATH=$OPENRESTY_PREFIX/nginx/sbin:$OPENRESTY_PREFIX/luajit/bin:$OPENRESTY_PREFIX/bin:$PATH
-    export OPENSSL111_BIN=$OPENRESTY_PREFIX/openssl111/bin/openssl
+    export OPENSSL_PREFIX=$OPENRESTY_PREFIX/openssl3
+    export OPENSSL_BIN=$OPENSSL_PREFIX/bin/openssl
```
Comment (monkeyDluffy6017) [inline ci/common.sh:29]: Not openssl3?

File: .github/workflows/link-check.yml [hunk 3]
```diff
@@ -1,49 +0,0 @@
-name: 'Link Checker'
```
Comment (monkeyDluffy6017) [inline .github/workflows/link-check.yml:1]: Why delete this?

Comment (AlinsRan) [reply .github/workflows/link-check.yml]: My mistake, there was a mistake here.

File: ci/common.sh (hunk 2 above)
Comment (AlinsRan) [reply ci/common.sh:29]: Openssl3 is the directory, and openssl is a binary program.

File: .github/workflows/cli-master.yml (hunk 1 above)
Comment (AlinsRan) [reply .github/workflows/cli-master.yml:13]: This CI is used to test whether the Rockspec for **remotely pulling** the master branch can be installed.
It is meaningless for PR, assuming that a third-party library in Rockspec has been updated and a certain field has been adjusted, then PR cannot pass.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by soulbird — APPROVED** [review]
Nice~~~

**Pull request merged**